### PR TITLE
Feature/users/sign up

### LINKED ISSUE #1: As a user, I want to be able to sign up for the app
"Hi! I don't know what this App is all about, but I really want to sign up for it!"

- backend
  - Add Spring Security, setup `SecurityFilterChain`
  - GET endpoint `/api/csrf` to retrieve CSRF token
  - POST endpoint `/api/users/signup`
  - `User` model
- frontend
  - signup page
  - route: `/signup`
  - navigation link
- unit and integration tests

### PR DESCRIPTION
**Pull request by valfink**
Resolve #1

### CODE REVIEW

File: backend/src/test/java/com/github/valfink/backend/mongouser/MongoUserControllerTest.java
Comment (gadmel) [inline backend/src/test/java/com/github/valfink/backend/mongouser/MongoUserControllerTest.java:1]: No tests for BadCredentialsExceptions? You throw them in 5 cases (at 3 places: two for bad username - null and empty string, the same two for password and one for username is already taken) I would recommend to write at least 3 tests for exceptions: one for each case and I would also advise to use different exceptions for bad credentials (403) and for already used (409).

Comment (gadmel) [reply backend/src/test/java/com/github/valfink/backend/mongouser/MongoUserControllerTest.java:1]: Sorry, I see now that you test in the service.. my bad

File: backend/src/test/java/com/github/valfink/backend/mongouser/MongoUserServiceTest.java
Comment (gadmel) [inline backend/src/test/java/com/github/valfink/backend/mongouser/MongoUserServiceTest.java:1]: Very very good tests. LVGTM!

File: backend/src/test/java/com/github/valfink/backend/util/CsrfControllerTest.java
Comment (gadmel) [inline backend/src/test/java/com/github/valfink/backend/util/CsrfControllerTest.java:1]: I know where do you have this one from 😉

File: frontend/src/App.tsx [hunk 1]
```diff
@@ -2,15 +2,31 @@ import React from 'react';
 import './App.css';
 import {Navigate, Route, Routes} from "react-router-dom";
 import FoodItemGallery from "./page/FoodItemGallery";
+import UserSignUpPage from "./page/UserSignUpPage";
+import axios from "axios";
+import Cookies from "js-cookie";
+
+axios.interceptors.request.use(config => {
+    if (["put", "post", "delete"].includes(config.method || "")) {
```
Comment (gadmel) [inline frontend/src/App.tsx:10]: A specified axios interceptor? - Interesting!

File: frontend/src/component/UserForm.tsx [hunk 2]
```diff
@@ -0,0 +1,51 @@
+import {Key, Person} from "react-bootstrap-icons";
```
Comment (gadmel) [inline frontend/src/component/UserForm.tsx:1]: Cool. Also take a look at these great [icons](https://fontawesome.com/).

**Review by gadmel — APPROVED** [review]
Great job, Valentin! Great job! 🐕

File: frontend/src/App.tsx (hunk 1 above)
Comment (valfink) [reply frontend/src/App.tsx:10]: Elvedin showed this shortly and I think it is very elegant. So you don't have to request a new CSRF cookie before you do a normal GET. Should save time and make the App faster :)

File: frontend/src/component/UserForm.tsx (hunk 2 above)
Comment (valfink) [reply frontend/src/component/UserForm.tsx:1]: Thanks for the tip! I will check them out :)

**Pull request merged**